com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -676,21 +697,19 @@ private void OnTransformParentChanged()
             m_IsReparented = true;
             ApplyNetworkParenting();
 
-            using (var buffer = PooledNetworkBuffer.Get())
-            {
-                using (var writer = PooledNetworkWriter.Get(buffer))
-                {
-                    writer.WriteUInt64Packed(NetworkObjectId);
-                    WriteNetworkParenting(writer, m_IsReparented, m_LatestParent);
-                }
+            var context = NetworkManager.MessageQueueContainer.EnterInternalCommandContext(
+                MessageQueueContainer.MessageType.ParentSync,
+                NetworkChannel.Internal,
+                NetworkManager.ConnectedClientsIds.Where((id) => Observers.Contains(id)).ToArray(),
```

Comment:
Linq `.Where` and `.ToArray` allocates and this is technically a hot path. Not sure whether it needs fixing now just pointing that we introduce "unnecessary" allocations here.

Reply:
What would be the better option here? We should definitely avoid unnecessary allocations. Would creating a new array and iterating the old one to manually populate the new one be better? I'm not sure if `new ulong[]` allocates.

maybe `stackalloc ulong[]` and iterate?

Reply:
I have put some thought into this and have determined that I know how I want to address it, but I'm going to make it part of my serialization RFC since one of the two primary reasons for that RFC is to remove allocations.